woocommerce/woocommerce
# Handle Invalid Product ID

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR fixes a problem after deleting a product selected for a `Single Product` block.
There is no way to Choose another product when the product has been deleted.

### Before

![Screenshot 2024-12-26 at 11 39 53 AM](https://github.com/user-attachments/assets/bdacd0f1-d675-499e-be4f-aba63c8cc155)

### After

![Screenshot 2024-12-30 at 4 39 11 PM](https://github.com/user-attachments/assets/bfd5383a-78c4-4004-aa18-fb4f6c55e346)

Closes #52662.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a product.
2. Go to Pages > Add New Page
3. Add a `Single Product` block and select the product you just created.
4. Save the page.
5. Remove the product and delete it permanently.
6. Reload the page.
7. The product picker should be visible with the text: `Previously selected product is no longer available.`
8. Pick another product.
9. The new product should be visible now.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
